R/fwrite.R

```diff
@@ -38,13 +41,17 @@ fwrite <- function(x, file="", append=FALSE, quote="auto",
     dec != sep,  # sep2!=dec and sep2!=sep checked at C level when we know if list columns are present
     is.character(eol) && length(eol)==1L,
     length(qmethod) == 1L && qmethod %chin% c("double", "escape"),
+    length(compress) == 1L && compress %chin% c("none", "gzip"),
     isLOGICAL(col.names), isLOGICAL(append), isLOGICAL(row.names),
     isLOGICAL(verbose), isLOGICAL(showProgress), isLOGICAL(logical01),
     length(na) == 1L, #1725, handles NULL or character(0) input
     is.character(file) && length(file)==1L && !is.na(file),
     length(buffMB)==1L && !is.na(buffMB) && 1L<=buffMB && buffMB<=1024,
     length(nThread)==1L && !is.na(nThread) && nThread>=1L
     )
+  
+  is_gzip <- compress == "gzip" || grepl("\\.gz$", file)
```

Comment:
Nice to recognise filename but we could warn if user explicitly set compress to none and uses gz filename

Reply:
Might be better "auto" instead of default. Anyway new argument is not required as we might do:
```r
is_gzip <- compress == "gzip" || (missing(compress) && grepl("\\.gz$", file))
```